lib/scanner.l

```diff
@@ -167,16 +168,31 @@ include_open      ^[ \t]*@include[ \t]+\"
 {float}           { yylval->fval = atof(yytext); return(TOK_FLOAT); }
 {integer}         {
                     long long llval;
-                    llval = atoll(yytext);
+                    char *endptr;
+                    int errsave = errno;
+                    errno = 0;
+                    llval = strtoll(yytext, &endptr, 0);	/* base 10 or base 8 */
+                    if(*endptr || errno)
+                    {
+                      errno = 0;
+                      return(TOK_ERROR);	/* some error occured ... */
+                    }
+                    errno = errsave;
+                    if((*yytext == '0') && (*(yytext+1) != '\0'))
```

Comment:
One more thing occurred to me last night about this change. I think it'd be simpler to declare a separate token for octals:

octal    0[1-7][0-7]*

Then all this error checking logic here wouldn't be necessary; you'd simply have a rule:

{octal}   { yylval->llval = strtoul(yytext, &endptr, 8); return(TOK_OCTAL); }

Very similar to the rule for {hex}.

We could also add support for binary literals in similar fashion.

{binary}   0b[0-1]+